src/OpenTelemetry.Instrumentation.Http/Implementation/HttpHandlerDiagnosticListener.cs

```diff
@@ -115,16 +111,54 @@ public void OnStartActivity(Activity activity, object payload)
                 return;
             }
 
+            bool isNullActivitySource = string.IsNullOrEmpty(activity.Source.Name);
+
             // Propagate context irrespective of sampling decision
             var textMapPropagator = Propagators.DefaultTextMapPropagator;
             if (textMapPropagator is not TraceContextPropagator)
             {
-                textMapPropagator.Inject(new PropagationContext(activity.Context, Baggage.Current), request, HttpRequestMessageContextPropagation.HeaderValueSetter);
+                var context = activity.Context;
+
+                // Note: isNullActivitySource is true for all .NET 6 and prior
+                // spans. isNullActivitySource is true for .NET 7+ only when
+                // sample decision was drop.
+                Debug.Assert(
+                    isNullActivitySource || IsNet7OrGreater,
+                    "activity source was invalid");
+
+                // Note: When sample decision is drop...
+                //  * On .NET 7 activity has empty source and is NOT updated by
+                //    the legacy logic. Recorded is just mirrored from the
+                //    parent.
+                //
+                //  * On .NET 6 activity is updated by the legacy logic and Recorded can be trusted.
+                if (isNullActivitySource
```

Comment:
I see one issue with this: Let's say I was using default TextmapPropagator i.e Trace and Baggage. Now, If I decide to stop using Baggage propagator and set my default as TraceContextPropagator, my end-to-end result will be different because then this will not be executed and HttpClient will simply use current activity for context propagation.

Reply:
I had to read that a few times but I got you 😄 Good point, pushed a fix.

Reply:
Ended up reverting this because we no longer mess with context.